src/components/ApplicationRoutes/ApplicationRoutes.tsx

```diff
@@ -96,13 +113,18 @@ export const ApplicationRoutes = (): React.ReactElement => {
   return (
     <>
       {serverError && <ErrorView {...serverError} />}
-      {!isLoggedIn && location.pathname !== AUTH_PATH && <Redirect to={getLoginPath()} />}
+      {!isLoggedIn && location.pathname !== AUTH_PATH && (
+        <LogoutView redirectPathname={previousPathname} />
+      )}
```

Comment:
Почему просто так не сделать?
```suggestion
      <Route path="/logout">
        <LogoutView redirectPathname={previousPathname} />
      </Route>
```